trin-core/src/portalnet/find/query_tracer.rs

```diff
@@ -0,0 +1,144 @@
+use crate::portalnet::Enr;
+
+use serde::Serialize;
+use std::collections::HashMap;
+use std::time::{Duration, SystemTime};
+
+///Store ENRs as strings for serialization simplicity
+type ENRString = String;
+
+/// Keeps track of query details.
+#[derive(Debug, Serialize)]
+pub struct QueryTracer {
+    /// None if the content was not found.
+    pub found_content_at: Option<ENRString>,
+    /// The local node.
+    pub origin: ENRString,
+    /// Map of a node's ID to its corresponding `QueryResponse`
+    pub responses: HashMap<ENRString, QueryResponse>,
+    /// Time that the query started.
+    #[serde(skip_serializing)]
+    pub started_at: SystemTime,
+}
```

Comment:
A few comments on this struct.

I would suggest renaming to `QueryTrace`.

I am generally opposed to JSON-RPC types outside of the `jsonrpc` module. I would prefer that the API object types (and the logic to convert to them from "core" types) only exist in the `jsonrpc` module. That may mean some duplication, but it enables us to change the internal representations without changing the API response.

`found_content_at` is not as clear as it could be. It could be interpreted as the time at which we found the content, which would be a meaning consistent with the `started_at` field.

The `origin` field is redundant because we know our own ENR.

Reply:
> I would suggest renaming to QueryTrace.
Agreed. 👍
> I am generally opposed to JSON-RPC types outside of the jsonrpc module. I would prefer that the API object types (and the logic to convert to them from "core" types) only exist in the jsonrpc module. That may mean some duplication, but it enables us to change the internal representations without changing the API response.

I'd prefer we handle any decoupling of types on an as-needed basis in the hypothetical future PR that makes it necessary. For now I think the `peertest` is sufficient for not accidentally changing API responses while changing internal representations.

> found_content_at is not as clear as it could be.
Agreed. 👍 I went with `received_content_from_node` which feels like a bit much. Any better ideas?

> The origin field is redundant because we know our own ENR.
Yea, but whoever's parsing this data needs to get the `origin` node's ENR at some point, and I think it should be bundled with the rest of the data rather than requiring a separate RPC call.